pkg/mesh/envoy.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"syscall"
+
+	"github.com/creack/pty"
+)
+
+const envoyUID = 1337
+const envoyGUID = 1337
+
+func (kr *KRun) envoyCommand() *exec.Cmd {
+	return exec.Command("/usr/local/bin/envoy",
+		"--config-path", fmt.Sprintf("%s/bootstrap.yaml", kr.TdSidecarEnv.PackageDirectory),
+		"--log-level", kr.GetTrafficDirectorLogLevel(),
+		"--log-path", "/var/log/envoy/envoy.log",
+		"--allow-unknown-static-fields",
+	)
+}
+
+// StartEnvoy does iptables interception, envoy bootstrap preparation and
+// runs envoy.
+func (kr *KRun) StartEnvoy() error {
```

Comment:
I guess we'll eventually stop using pilot-agent and do most of the work ( to reduce binary size).